nft_ingester/src/program_transformers/token_metadata/v1_asset.rs

```diff
@@ -98,78 +88,40 @@ pub async fn save_v1_asset<T: ConnectionTrait + TransactionTrait>(
-                .column_as(token_accounts::Column::Owner, "owner")
-                .column_as(token_accounts::Column::Delegate, "delegate")
-                .join(
-                    JoinType::InnerJoin,
-                    tokens::Entity::belongs_to(token_accounts::Entity)
-                        .from(tokens::Column::Mint)
-                        .to(token_accounts::Column::Mint)
-                        .into(),
-                )
-                .into_model::<OwnershipTokenModel>()
-                .one(conn)
-                .await?;
+    // gets the token and token account for the mint to populate the asset. This is required when the token and token account are indexed, but not the metadata account. If the metadata account is indexed, then the token and ta ingester will update the asset with the correct data
 
-            Ok(result.map(|t| {
-                let token = tokens::Model {
-                    mint: t.mint.clone(),
-                    supply: t.supply,
-                    //Not Needed here
-                    decimals: 0,
-                    token_program: vec![],
-                    mint_authority: None,
-                    freeze_authority: None,
-                    close_authority: None,
-                    extension_data: None,
-                    slot_updated: 0,
-                };
-                let token_account = token_accounts::Model {
-                    pubkey: vec![],
-                    mint: t.mint,
-                    owner: t.owner,
-                    amount: t.token_account_amount,
-                    delegate: t.delegate,
-                    //Not Needed here
-                    frozen: false,
-                    close_authority: None,
-                    delegated_amount: 0,
-                    slot_updated: 0,
-                    token_program: vec![],
-                };
-                (token, Some(token_account))
-            }))
-        }
-        _ => {
-            let token = tokens::Entity::find_by_id(mint.clone()).one(conn).await?;
-            Ok(token.map(|t| (t, None)))
+    let (token, token_account): (Option<tokens::Model>, Option<token_accounts::Model>) =
+        match ownership_type {
+            OwnerType::Single => {
+                let token: Option<tokens::Model> =
+                    tokens::Entity::find_by_id(mint.clone()).one(conn).await?;
+                // query for token account associated with mint with positive balance
+                let token_account: Option<token_accounts::Model> = token_accounts::Entity::find()
+                    .filter(token_accounts::Column::Mint.eq(mint.clone()))
+                    .filter(token_accounts::Column::Amount.gt(0))
+                    .one(conn)
+                    .await?;
+                Ok((token, token_account))
+            }
+            _ => {
+                let token = tokens::Entity::find_by_id(mint.clone()).one(conn).await?;
+                Ok((token, None))
+            }
```

Comment:
Looks like `OwnerType::Single` is for NFTs and `OwnerType::Token` is for fungible assets and fungible tokens.  But wouldn't those also have the same kind of `token_account` to parse as with NFTs, just with supply > 1 and/or decimals > 0?

Reply:
You'll have N token_accounts. So you can't find one.

Reply:
Oh yeah that makes sense I didn't think it through properly.